File: sum_of_all_multiples.rb
def find(n)
 (1..n).to_a.select { |num| num % 3 == 0 || num % 5 == 0 }.reduce(:+)
end

def find(n)
 (3..n).select { |num| num % 3 == 0 || num % 5 == 0 }.sum
end

def find(n)
 (3..n).reduce { |sum, num| num % 3 == 0 || num % 5 == 0 ? sum + num : sum}
end

def find(n)
 [*3..n].reduce { |sum, num| num % 3 == 0 || num % 5 == 0 ? sum + num : sum}
end

find(5)

File: sumOfAllMultiples.js
//Sum of all the multiples of 3 or 5 (7 kyu)
function findSum(n) {
  let sumNums = 0;
  for (let i = 0; i <= n; i++) {
    if ((i % 3 === 0) || (i % 5 === 0)) {
      sumNums += i;
    }
  }
  return sumNums;
}

function findSum(n) {
  return [...Array(n + 1).keys()]
  .filter(key => ((key % 3 === 0) || (key % 5 === 0)))
  .reduce((sum, num) => sum + num);
}

function findSum(n) {
 if (n < 3) return 0
 return (n % 3 === 0 || n % 5 === 0) ? n + findSum(n-1) : findSum(n-1)
}

findSum(5);
